### PR TITLE
Take into account the proba threshold for binary classif

### PR DESCRIPTION
**Pull request by du-phan**
When dealing with binary classification, we take into the account used by user/doctor.

This is in relation with [this PR on MEA plugin](https://github.com/dataiku/dss-plugin-model-error-analysis/pull/20)

### CODE REVIEW

**Review by simonamaggio — APPROVED** [review]

File: mealy/error_analyzer.py [hunk 1]
```diff
@@ -240,7 +242,16 @@ def _compute_primary_model_error(self, X, y):
...
-        y_pred = self._primary_model.predict(X)
+        if is_regressor(self._primary_model):
+            y_pred = self._primary_model.predict(X)
+        else: # classification model
+            if len(np.unique(y)) > 2: # multiclass -> nothing to do
+                y_pred = self._primary_model.predict(X)
+            else: # binary -> need to check the proba threshold
+                prediction_index = (self._primary_model.predict_proba(X)[:, 1] > self.probability_threshold).astype(int)
+                #map the prediction indexes to the original target values
+                y_pred = [self._primary_model.classes_[i] for i in prediction_index]
```
Comment (AgatheG) [inline mealy/error_analyzer.py:253]: - Nitpick:
```suggestion
        if is_regressor(self._primary_model) or len(np.unique(y)) > 2:
           # Regression or multiclass classification models: no proba threshold
            y_pred = self._primary_model.predict(X)
        else:
            # Binary classification model
            prediction_index = (self._primary_model.predict_proba(X)[:, 1] > self.probability_threshold).astype(int)
            # Map the prediction indexes to the original target values
            y_pred = [self._primary_model.classes_[i] for i in prediction_index]
```

- Also small remark, for binary classif we end uup with a liist, but in other casses we end up with an array-like (depending on X type, can be a np.array). Ensure if this matters

**Review by AgatheG — COMMENTED** [review]
Did not test the code,two minor remarks

File: mealy/error_analyzer.py (hunk 1 above)
Comment (du-phan) [reply mealy/error_analyzer.py:253]: @AgatheG which variable are you talking about ? `y_pred` ?

Comment (du-phan) [reply mealy/error_analyzer.py:253]: Good point, for classif the variable is only used here https://github.com/dataiku/mealy/blob/fix/add-proba-threshold/mealy/error_analyzer.py#L285 , which works between list and array.

Comment (du-phan) [reply mealy/error_analyzer.py]: I casted it to np.array in d9ee24a anw

**Review by AgatheG — APPROVED** [review]

**Pull request merged**